### ISSUE TITLE
[UX] Fix `outside-right` styles so text doesn't overlap

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#6687]
As reported by @izmeez and further tracked down by @avpaderno, at tablet-ish screen sizes the admin menu fly outs too close to the right side of the screen get text overlap.

![Screenshot of Date and Time Formats Flyout overlapping dropdown](https://private-user-images.githubusercontent.com/6208775/344317181-06144314-289f-42d9-bed3-9dcc7fbeced9.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwMDkyMjUsIm5iZiI6MTcyNDAwODkyNSwicGF0aCI6Ii82MjA4Nzc1LzM0NDMxNzE4MS0wNjE0NDMxNC0yODlmLTQyZDktYmVkMy05ZGNjN2ZiZWNlZDkuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MThUMTkyMjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTY0M2EzZjNjODllMGQxZmY5NjlmZDA5OTlhMzg0ZmRiN2ViMzcyY2JiM2NjYTU4MDM1MGEyYWY0MjJkYzY2NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xPQZT5DEx8ugOTx547DRsSkZ-MjkV2PVKX27SGP9NPE)

I found the issue is with the `outside-right` styling, will be sending a PR shortly

## Steps To Reproduce

To reproduce the behavior:
1. Login so the admin bar is visible
2. Adjust screen width to roughly 1250px
3. Hover over Configuration > Regional and Language > Date and time format
4. This should cause the sub-menu to appear on top of the parent menu item, overlapping it. It can be difficult to distinguish the overlapping sub-menu from the parent menu.

**Comment by wesruv** [comment]
Here is what my suggest fix looks like:
![image](https://github.com/user-attachments/assets/48d821b0-fe89-4a61-b140-79c95aca39b9)

Adding a different background color, a generous drop shadow, and a border (that's actually a box shadow so the style mess with layout).

I'm not completely sold on the look, but I think it works.

I considered making the dropdown a different width to make it more obvious, but I'm pretty sure that'd open a can of worms with how flyout menus are handled.

**Comment by indigoxela** [comment]
This is what we have now - to directly compare with the change:

![fly-out-right-before-after](https://github.com/user-attachments/assets/79e512a7-4f17-47ea-88fa-5e9ae8a44116)

And that works like a charm. :+1: 

What I find a bit irritating is, that for RTL languages, the box shadow _always_ appears, even if far from the window edge.

![shadow-rtl-always-there](https://github.com/user-attachments/assets/c45d945e-229a-4a41-af6c-a768959d93c8)

What's the intention behind that?

**Comment by stpaultim** [comment]
@wesruv - This is an improvement, but I'm not sure if I would have noticed the improvement if I weren't looking for it. To my eyes it was not that visible. Could we do something a bit more dramatic?

**Comment by quicksketch** [comment]
> What I find a bit irritating is, that for RTL languages, the box shadow always appears, even if far from the window edge (and no overlap).

Nice catch @indigoxela. I think the newly added RTL CSS might have been added to the wrong selector. I posted a suggestion here: https://github.com/backdrop/backdrop/pull/4847/files#r1739451230

I think this is a nice little improvement. Also, I think it's a UX improvement and could be made post feature-freeze.

**Comment by jenlampton** [comment]
> Adding a different background color, a generous drop shadow, and a border (that's actually a box shadow so the style mess with layout).

The PR seems to only have box-shadow changes.  I'm not seeing the change in background color, as in @wesruv 's first screenshot:
![image](https://github.com/user-attachments/assets/7152bbd3-3367-48f1-a6a9-d04299bdabc4)

I see the darker black-gray which is not quite as helpful:
<img width="485" alt="Screenshot 2024-09-12 at 1 09 15 PM" src="https://github.com/user-attachments/assets/d16d3361-ed51-4436-89d1-ed1fd0f9e6b2">

Did we decide against the background color? It's possible that could warrent a larger discussion, since right now every item with a child gets a distinct color, and the suggestion above would be using that color to indicate soemthing other than "has a child link".  I'm not sure the current pattern is a good pattern, but that's for another issue :)

**Comment by jenlampton** [comment]
I think we have some other work to do on the colors, so perhaps saving that for later is good. Here's what other menu links look like with the box-border changes only:

Text formats:
<img width="446" alt="Screenshot 2024-09-12 at 1 22 39 PM" src="https://github.com/user-attachments/assets/11ca4b33-358a-4937-881c-f454dfc88b80">

Configuration manager:
<img width="412" alt="Screenshot 2024-09-12 at 1 22 29 PM" src="https://github.com/user-attachments/assets/48389993-8482-46c5-ad04-804c0694db33">

URL aliases:
<img width="447" alt="Screenshot 2024-09-12 at 1 22 05 PM" src="https://github.com/user-attachments/assets/345d1172-4ec0-4e92-a8f8-0e2141c0aee0">

Image styles:
<img width="447" alt="Screenshot 2024-09-12 at 1 21 45 PM" src="https://github.com/user-attachments/assets/c971a5b6-064f-4eb2-b20f-b7f22d367302">

And the original Date formats:
<img width="485" alt="Screenshot 2024-09-12 at 1 09 15 PM" src="https://github.com/user-attachments/assets/6a34fecd-1dfb-40f5-b9ac-7e57cf3665b6">

**Comment by jenlampton** [comment]
I've also re-applied the RTL changes as @quicksketch recommended, and here's what that looks like now:

Text formats:
<img width="508" alt="Screenshot 2024-09-12 at 1 27 21 PM" src="https://github.com/user-attachments/assets/1b317f84-6977-495a-8977-bfafeb9d3080">

Configuration manager:
<img width="518" alt="Screenshot 2024-09-12 at 1 27 29 PM" src="https://github.com/user-attachments/assets/e598b489-a95f-4b40-b844-8b51e1481ad3">

URL aliases:
<img width="547" alt="Screenshot 2024-09-12 at 1 27 37 PM" src="https://github.com/user-attachments/assets/ff8c6e20-04f2-4528-b7f7-f6dc1668e6f8">

Image styles:
<img width="522" alt="Screenshot 2024-09-12 at 1 29 55 PM" src="https://github.com/user-attachments/assets/51768b31-8541-434b-bcc3-b3d876790d64">

And the original Date formats:
<img width="508" alt="Screenshot 2024-09-12 at 1 27 13 PM" src="https://github.com/user-attachments/assets/22d5f079-7179-47f1-be72-c4b108339133">

**Comment by quicksketch** [comment]
Confirmed @jenlampton's changes work as expected. The additional dropshadow and border are added only after the menu begins to wrap, both in LTR and RTL languages.

**Comment by quicksketch** [comment]
I merged https://github.com/backdrop/backdrop/pull/4869 into 1.x for 1.29.0. Thank you @wesruv for creating the issue and filing the initial PR!

https://github.com/backdrop/backdrop/commit/5aa692b454197e25ef0936e8c74550b1e6b59966 by @wesruv, @jenlampton, @izmeez, @indigoxela, and @quicksketch.
